modules/ROOT/pages/installing-kubernetes.adoc

```diff
@@ -0,0 +1,264 @@
+
+For example, to deploy a cluster with 4 regular nodes, 1 import node and 1 replicator node you would run the steps below 3 times, each time with a different configuration and deployment controller file.
+
+=== Configuration file
+
+In this section, you will create the Sync Gateway configuration file and store it in a secret location.
+
+. Create a new file called *sgw-config.json* with the following and replace the `server` key with the addressable Couchbase Server hostname.
++
+[source,json]
+----
+{
+  "logging": {
+    "console": {
+      "log_keys": ["*"]
+    }
+  },
+  "databases": {
+    "db": {
+      "server": "http://cb-example-0000.cb-example:8091", // <1>
+      "bucket": "default",
+      "username": "admin", // <2>
+      "password": "password",
+      "users": { "GUEST": { "disabled": false, "admin_channels": ["*"] } },
+      "allow_conflicts": false,
+      "revs_limit": 20
+    }
+  }
+}
+----
+<1> The server key should point to the Couchbase Server cluster.
+This would typically be of the form `http://CB_SERVER_POD.CB_SERVER_SERVICE_NAME:8091`.
+<2> The username/password keys should match what was setup in the <<pre-requisites, pre-requisites>> section when you configured the Sync Gateway RBAC user.
+. You will use a https://kubernetes.io/docs/tasks/configure-pod-container/configure-pod-configmap/[Kubernetes configMap] to pass the configuration file to Sync Gateway on launch.
+However, since the Sync Gateway contains sensitive information, it is recommended that you create a secret from the config file and pass that to the Sync Gateway.
+Run the following command to create a secret called "sgw-config" from the specified Sync Gateway configuration file (*sgw-config.json*).
++
+[source,console]
+----
+kubectl create secret generic sgw-config --from-file sgw-config.json
+----
++
+If successful, you will see the following.
++
+[source,console]
+----
+secret "sgw-config" created
+----
+
+=== Deployment controller
+
+In this section you will deploy the Sync Gateway cluster with the configuration file that you created above.
+
+You will be using a https://kubernetes.io/docs/concepts/workloads/controllers/deployment/[Kubernetes deployment controller].
+A deployment controller allows you to define the number of Sync Gateway replicas and other parameters.
+
+. Create a new file called *sgw-deployment.yaml* with the following.
++
+[source,yaml]
+----
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: sync-gateway <1>
+spec:
+  replicas: 2 <2>
+  template:
+    metadata:
+      labels:
+        app: sync-gateway
+    spec:
+      containers:
+        - name: sync-gateway
+          image: couchbase/sync-gateway:2.1.2-enterprise <3>
+          args: ["/sync-gateway-config/sgw-config.json"] <4>
+          volumeMounts:
+            - name: sgw-config-volume
+              mountPath: /sync-gateway-config
+              readOnly: true
+          env:
+            - name: GOMAXPROCS
```

Comment:
Don't know if this needs a comment about this limiting Sync Gateway to using 2 CPU?
If this is not set it will create as many processes as CPU cores on the host machine, so is important to bear in mind in a Kubernetes environment.

Reply:
Can you clarify why  "2" the magic number ?( I think it makes sense to specify what this config does in general . Recommend 2 and then leave it to user.)

*That said*, I think I have to check if we need to restrict SGW pod to a VM or node similar to server- So let me confirm that first. Because if we have anti affinity that we do not need this restriction.

Reply:
Yeah, that's what I meant, the actual idea behind the environment variable (rather than the value it's set to, that's pretty arbitrary).

You can add anti-affinity in the deployment, but it is only anti-affinity to the other pods in that deployment. So in theory if you had different deployments with an import node and regular Sync Gateways, these could be scheduled on the same underlying Kubernetes node.
I think it makes sense to try and limit Sync Gateways to running on their own dedicated nodes, but without specifically labelling nodes and ensuring nothing else runs on those besides Sync Gateway, the only way to ensure Sync Gateway 'plays nice' with whatever else is scheduled on that node is to limit its resource and set its GOMAXPROCS (so it knows it has limited resource).

Reply:
Agree. Maybe we could think about a best practices section similar to CBS ( separate PR). 
@jamiltz  Please update text to clarify the purpose of cpus
- cpu : This is used to specify the CPU limit for the Sync Gateway pod. If you do not specify one, the Sync Gateway could spawn as many processes as CPU cores and potentially use up all CPU resources. You can learn more about CPU resource assignment [here](https://kubernetes.io/docs/tasks/configure-pod-container/assign-cpu-resource/#if-you-do-not-specify-a-cpu-limit). We recommend a value of 2 but you should use what is suited for your environment
- GOMAXPROCS : This GO runtime env variable is used to limit the number of system threads that are allocated to the Sync Gateway

Reply:
https://github.com/couchbase/docs-sync-gateway/commit/9189d889bd4efaabd09115b93e17977d8f9a385d